### ISSUE TITLE
Feature request: Tab to switch fields in context menu

### ISSUE THREAD

**Issue by EfremBraun** [pyqtgraph/pyqtgraph#1687]
In the right-click context menu, entering the axis limits manually lends itself to tabbing to switch between the min and the max fields, but currently, clicking tab just removes focus. It would be nice to implement this feature if possible.

**Comment by j9ac9k** [comment]
I agree, this would be good to have.  I don't remember how to set tab order off the top of my head, but that's well defined Qt behavior in general if memory serves.

**Comment by EfremBraun** [comment]
> I agree, this would be good to have. I don't remember how to set tab order off the top of my head, but that's well defined Qt behavior in general if memory serves.

I gave it a try, but no success for now.

Tab order is well defined, see https://doc.qt.io/archives/qtforpython-5.12/PySide2/QtWidgets/QWidget.html#PySide2.QtWidgets.PySide2.QtWidgets.QWidget.setTabOrder.

However, it looks like ViewBoxMenu is a QMenu widget, which unfortunately is not focusable per https://stackoverflow.com/a/15968275/. Per https://stackoverflow.com/a/20388856, this can be overridden, but my attempt to do this didn't succeed.

Tried so far (all unsuccessfully):
- Setting tab orders in axisCtrlTemplate_pyside2.py (via `QtWidgets.QWidget.setTabOrder(self.minText, self.maxText)` in setupUi())
- Setting focus policy in axisCtrlTemplate_pyside2.py (via `Form.setFocusPolicy(QtCore.Qt.StrongFocus)` in setupUi())
- Overriding focusNextPrevChild() in ViewBoxMenu.py per https://stackoverflow.com/a/20388856

**Comment by j9ac9k** [comment]
Thanks for giving it a go; ahh yeah if QMenu isn't focusable this gets a bit more complicated.  I'll take off the good first issue label.

**Comment by bbc131** [comment]
I just stumbled over the same problem and want to share the following finding: 
Using a `QWidgetAction` for each widget in a `QMenu` seems to solve the problem. 
Currently all widgets in a menu are embedded within a single `QWidgetAction`.

**Comment by EfremBraun** [comment]
> I just stumbled over the same problem and want to share the following finding: Using a `QWidgetAction` for each widget in a `QMenu` seems to solve the problem. Currently all widgets in a menu are embedded within a single `QWidgetAction`.

@bbc131: Could you show how https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/graphicsItems/ViewBox/ViewBoxMenu.py would be adjusted to fix this? I just tried changing the `for` loop in lines 29-57 to:
```
for axis in 'XY':
            m = QtWidgets.QMenu()
            m.setTitle(f"{axis} {translate('ViewBox', 'axis')}")
            w = QtWidgets.QWidget()
            ui = ui_template.Ui_Form()
            ui.setupUi(w)
            connects = [
                (ui.mouseCheck.toggled, 'MouseToggled'),
                (ui.manualRadio.clicked, 'ManualClicked'),
                (ui.minText.editingFinished, 'RangeTextChanged'),
                (ui.maxText.editingFinished, 'RangeTextChanged'),
                (ui.autoRadio.clicked, 'AutoClicked'),
                (ui.autoPercentSpin.valueChanged, 'AutoSpinChanged'),
                (ui.linkCombo.currentIndexChanged, 'LinkComboChanged'),
                (ui.autoPanCheck.toggled, 'AutoPanToggled'),
                (ui.visibleOnlyCheck.toggled, 'VisibleOnlyToggled')
            ]
            for sig, fn in connects:
                a = QtWidgets.QWidgetAction(self)
                a.setDefaultWidget(w)
                m.addAction(a)
            self.addMenu(m)
            self.axes.append(m)
            self.ctrl.append(ui)
            wg = WidgetGroup(w)
            self.widgetGroups.append(wg)
            
            for sig, fn in connects:
                sig.connect(getattr(self, axis.lower()+fn))

```
but that didn't change the tab behavior.

**Comment by bbc131** [comment]
Maybe I wasn't clear, this would need some rework and I think it's not possible to get the same GUI layout within the menus as currently.

Let me explain this in detail:
Let's take the X axis submenu as an example. All elements in this submenu, which you can see in the screenshot, are contained in one and the same `QWidgetAction`, which contains one `QWidget`, with all the elements contained in a layout. 
![grafik](https://user-images.githubusercontent.com/36670201/178427885-3e798587-f8d1-4286-afe2-6fda6ed48218.png)

In the lines 
```
w = QtWidgets.QWidget()
ui = ui_template.Ui_Form()
ui.setupUi(w)
```
these elements are instantiated and placed in a single `QWidget`, which is then embedded via the `QWidgetAction` within the menu.

Now, what needs to be changed is the whole `ui_template` part. The elements must not be contained in one `QWidget`. Each element must be embedded in its own `QWidgetAction`.
To see, that this would enable the "tabbing" between the elements, you can add the following code.
```
...
             m = QtWidgets.QMenu()
             m.setTitle(f"{axis} {translate('ViewBox', 'axis')}")
+
+            foo = QtWidgets.QRadioButton("foo")
+            foo_wa = QtWidgets.QWidgetAction(self)
+            foo_wa.setDefaultWidget(foo)
+            m.addAction(foo_wa)
+
+            bar = QtWidgets.QSpinBox()
+            bar_wa = QtWidgets.QWidgetAction(self)
+            bar_wa.setDefaultWidget(bar)
+            m.addAction(bar_wa)
+
             w = QtWidgets.QWidget()
             ui = ui_template.Ui_Form()
...
```
With this, two elements are added before the existing elements, see the screenshot below. Each in its own `QWidgetAction`, which is why we can switch focus via Tab-key.
![grafik](https://user-images.githubusercontent.com/36670201/178427938-9efecd40-1c94-47c3-b07c-5d33adae7ac3.png)

Note, that since each `QWidgetAction` goes into a new line in the menu, we cannot put two GUI elements within the same line and be able to switch focus via Tab-Key between them.
